puzzler.py

```diff
@@ -63,16 +63,17 @@ def score_puzzle(info):
     return (min_depth + stable_depth) / 2
 
 
-def generate_puzzles(instream, outstream, engine, variant, multipv, depth, min_score):
+def generate_puzzles(instream, outstream, engine, variant, multipv, depth, min_score, evalfile):
     engine.setoption('UCI_Variant', variant)
     engine.setoption('multipv', multipv)
+    engine.setoption('EvalFile', evalfile)
     for fen in instream:
         fen = fen.strip().split(';')[0]  # also support EPD
         pv = []
         scores = []
         types = []
         while True:
-            puzzle_type, info = get_puzzle(variant, fen, pv, engine, depth)
+            puzzle_type, info = get_puzzle(variant, fen, pv, engine, depth, evalfile)
```

Comment:
The `get_puzzle` function doesn't have this argument.

Reply:
Sorry for my confusing statement. What I meant is that `get_puzzle` doesn't need this argument. Adding it in both places as you did also works, but it would be unused, so it is preferable to remove it in both places.